src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -767,28 +843,54 @@ private async Task<Dictionary<string, ConfigurationSetting>> LoadSelectedKeyValu
                     LabelFilter = LabelFilter.Null
                 };
 
+                List<MatchConditions> matchConditions = new List<MatchConditions>();
+
                 await CallWithRequestTracing(async () =>
                 {
-                    await foreach (ConfigurationSetting setting in client.GetConfigurationSettingsAsync(selector, cancellationToken).ConfigureAwait(false))
+                    AsyncPageable<ConfigurationSetting> pageableSettings = client.GetConfigurationSettingsAsync(selector, cancellationToken);
+
+                    await foreach (Page<ConfigurationSetting> page in _options.PageableManager.GetPages(pageableSettings).ConfigureAwait(false))
                     {
-                        serverData[setting.Key] = setting;
+                        foreach (ConfigurationSetting setting in page.Values)
+                        {
+                            existingData[setting.Key] = setting;
+                        }
+
+                        matchConditions.Add(new MatchConditions { IfNoneMatch = page.GetRawResponse().Headers.ETag });
                     }
                 }).ConfigureAwait(false);
+
+                watchedCollections[new KeyValueIdentifier(KeyFilter.Any, LabelFilter.Null)] = matchConditions;
             }
 
-            foreach (KeyValueSelector loadOption in _options.KeyValueSelectors)
+            foreach (KeyValueSelector loadOption in selectors)
             {
-                IAsyncEnumerable<ConfigurationSetting> settingsEnumerable;
-
                 if (string.IsNullOrEmpty(loadOption.SnapshotName))
                 {
-                    settingsEnumerable = client.GetConfigurationSettingsAsync(
-                        new SettingSelector
+                    var selector = new SettingSelector()
+                    {
+                        KeyFilter = loadOption.KeyFilter,
+                        LabelFilter = loadOption.LabelFilter ?? LabelFilter.Null
+                    };
+
+                    List<MatchConditions> matchConditions = new List<MatchConditions>();
+
+                    await CallWithRequestTracing(async () =>
+                    {
+                        AsyncPageable<ConfigurationSetting> pageableSettings = client.GetConfigurationSettingsAsync(selector, cancellationToken);
+
+                        await foreach (Page<ConfigurationSetting> page in _options.PageableManager.GetPages(pageableSettings).ConfigureAwait(false))
                         {
-                            KeyFilter = loadOption.KeyFilter,
-                            LabelFilter = loadOption.LabelFilter
-                        },
-                        cancellationToken);
+                            foreach (ConfigurationSetting setting in page.Values)
```

Comment:
I think we should check the response status code before anything else. What happens if the response is a failure? Does SDK still throw RequestFailedException even though we are managing each page ourselves?

Reply:
>  For 4xx and 5xx, I suppose SDK to throw exception

Have we tested this to make sure SDK will throw?

Reply:
Will test the behavior in .NET for whatever exceptions I can

Reply:
We need to check for .NET SDK too because the API is different for .NET than other SDKs @amerjusupovic

Reply:
I tested this with invalid endpoint/credentialto test some 4xx status codes and the exceptions are caught and handled as expected. On startup it catches the exceptions and waits until timeout.